proof/refine/InvariantsPre_H.thy

```diff
@@ -5,45 +5,76 @@
  * SPDX-License-Identifier: GPL-2.0-only
  *)
 
-theory Invariants_H
+(* Preliminary definitions required to define arch-specific invariants and properties.
+   These are primarily related to obj_at' and storability.
+*)
+
+theory InvariantsPre_H
 imports
   LevityCatch
   "AInvs.ArchDetSchedSchedule_AI"
   "Lib.Heap_List"
+  Move_R
 begin
 
 (* global data and code of the kernel, not covered by any cap *)
 axiomatization
   kernel_data_refs :: "word64 set"
 
-context Arch begin
 
-declare lookupPTSlotFromLevel.simps[simp del]
-declare lookupPTFromLevel.simps[simp del]
+section \<open>Locale Setup for Idempotence over kernel_state Field Updates\<close>
```

Comment:
Part of the confusion in the comment may be that "idempotence" is not the right word. Idempotence means `f (f x) = f x`, i.e. applying more than once is the same as applying once. What we are proving here is more identity, i.e. f x = x, but with side conditions. The locales are identifying which updates are the identity function on specific predicates. For instance, when a function doesn't change the `ksPSpace` component of the state, that function is the identity on all predicates that only depend on `ksPSpace`.

+1 on adding where these are populated and interpreted, I think that would be a good idea.

The setup is good, though, I'm happy with making this whole setup a bit more principled.

Reply:
> +1 on adding where these are populated and interpreted, I think that would be a good idea.

What do you mean by this?

Reply:
I assume that that was referring to my earlier [comment](https://github.com/seL4/l4v/pull/833#discussion_r1872729019), where I suggested "maybe with an explanation that the locales are actually populated and interpreted in Invariants_H".

Reply:
yeah, what Corey said. I didn't realise my comment would not show up under Corey's comment (it did in the review view, but apparently not here).

Reply:
Hopefully that's done, let me know if it explains sufficiently.

Reply:
The section title still has "Idempotence" in it

Reply:
Apologies, it does not, I was staring at the old version. Happy with the new version!